### PR TITLE
Quality enhancements

### PR DESCRIPTION
**Pull request by SalomonBrys**
This PR adds various quality enhancements:

- Constraints JitPack to only serving requery artifacts. This is a temporary measure. Jitpack needs to be removed from the requirements. Nikhil Purushe (Requery Sqlite-Android maintainer) has said on April 18th that he will look into publishing the lib to Maven Central "in the next couple of weeks". If that does not come through, Sqlite-Android will need to be forked.
- Enables [Explicit API mode](https://kotlinlang.org/docs/jvm-api-guidelines-backward-compatibility.html#explicit-api-mode).
- Enhances the SupabaseConnector to allow it to receive a SupabaseClient if the user already maintains one.
- Fixes some glitches in the documentation.
- Updates logging in order to use Android & iOS specific logging capabilities (through the Canard lib).
- Adds secondary vararg constructors to Index and Schema (to ease the schema DSL).
- Provide the `compose` artifact that offers `rememberDatabaseFactory` for Compose users.

### CODE REVIEW

**Review by stevensJourney — COMMENTED** [review]
Thanks so much for making this PR!

From my side I'm very happy with the enhancements. I've left some minor comments. Please let me know if you have any questions or need any assistance with them.

File: core/src/commonMain/kotlin/com/powersync/db/schema/IndexedColumn.kt [hunk 1]
```diff
@@ -26,28 +26,28 @@ data class IndexedColumn(
...
+        public fun ascending(column: String): IndexedColumn = IndexedColumn(column, true)
+        public fun descending(column: String): IndexedColumn = IndexedColumn(column, false)
     }
 
     /**
      * Sets the parent column definition. The column definition's type
      * is required for the serialized JSON payload of powersync_replace_schema
      */
-    fun setColumnDefinition(column: Column) {
+    public fun setColumnDefinition(column: Column) {
```
[the remark this answers is not in the document]
Comment (SalomonBrys) [reply core/src/commonMain/kotlin/com/powersync/db/schema/IndexedColumn.kt]: Done

File: core/src/commonMain/kotlin/com/powersync/sync/SyncStatus.kt [hunk 2]
```diff
@@ -89,15 +89,15 @@ data class SyncStatusDataContainer(
 }
 
 
-data class SyncStatus(
+public data class SyncStatus internal constructor(
```
[the remark this answers is not in the document]
Comment (SalomonBrys) [reply core/src/commonMain/kotlin/com/powersync/sync/SyncStatus.kt:92]: Added `SyncStatus.empty()` for this use case.

File: README.md [hunk 3]
```diff
@@ -195,17 +193,29 @@ val driverFactory = DatabaseDriverFactory()
...
 database.connect(MyConnector())
   ```
 
+**Special case: Compose Multiplatform**
+
+The artifact `com.powersync:powersync-compose` provides a simpler API:
+
+```kotlin
+// commonMain
+val database = rememberPowerSyncDatabase(schema)
```
[the remark this answers is not in the document]
Comment (SalomonBrys) [reply README.md:213]: Done

**Review by stevensJourney — COMMENTED** [review]
Thanks! Looks good to me. I'd just like to have @Manrich121's eyes on it before merging.

**Review by stevensJourney — APPROVED** [review]

**Pull request merged**